syntax = "proto3";

service grpc_plugin_obj {
  rpc com_grpc_js_api (grpc_js_api_Request) returns (grpc_js_api_Reply) {}
  rpc com_grpc_hello (grpc_hello_Request) returns (grpc_js_api_Reply) {}
  rpc com_grpc_js_executeCallback (grpc_js_api_ExecuteCallback) returns (grpc_js_api_Reply) {}
  rpc com_grpc_window_toggleVisibility (grpc_window_toggleVisibility) returns (grpc_empty_Reply) {}
}

service grpc_proxy_obj {
  rpc com_grpc_js_api (grpc_js_api_Request) returns (grpc_js_api_Reply) {}
  rpc com_grpc_js_executeCallback (grpc_js_api_ExecuteCallback) returns (grpc_js_api_Reply) {}
}

service grpc_browser_window_obj {
  rpc com_grpc_window_toggleVisibility (grpc_window_toggleVisibility) returns (grpc_empty_Reply) {}
}

// Client->
message grpc_hello_Request {
	int32 peerport = 1;
}

// Client->
message grpc_js_api_Request {
	string funcname = 1;
	string params = 2;
	int32 peerport = 3;
}

// Client->
message grpc_js_api_ExecuteCallback {
	int32 funcid = 1;
	string jsonstr = 2;
}

// Client->
message grpc_window_toggleVisibility {
	string empty = 1;
}

// Server->
message grpc_js_api_Reply {
	string empty = 1;
}

// Server->
message grpc_empty_Reply {
	string empty = 1;
}
